Tests/NumLib/FeTestData/TestFeQUAD8.h

```diff
@@ -0,0 +1,57 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *  \file   TestFeQUAD8.h
+ *  Created on July 13, 2017, 3:00 PM
+ */
+
+#pragma once
+#include "MeshLib/Elements/Quad.h"
+#include "NumLib/Fem/FiniteElement/C0IsoparametricElements.h"
+
+namespace FeTestData
+{
+class TestFeQUAD8
+{
+public:
+    using ShapeFunction = NumLib::ShapeQuad8;
+
+    // Fe type information
+    template <template <typename> class ShapeMatrixPolicy_>
+    using FeType = NumLib::FeQUAD8<ShapeMatrixPolicy_>;
+
+    using MeshElementType = MeshLib::Quad8;
+    static const unsigned dim = 2;  // MeshElementType::dimension;
+    static const unsigned e_nnodes = MeshElementType::n_all_nodes;
+    static const unsigned n_sample_pt_order2 = 2 * 2;
+    static const unsigned n_sample_pt_order3 = 3 * 3;
+    static const unsigned global_dim = MeshElementType::dimension;
+
+    const double perturbation = -0.1;
+
+    /// create a mesh element
+    MeshElementType* createMeshElement()
+    {
+        // square
+        auto** nodes = new MeshLib::Node*[e_nnodes];
+        nodes[0] = new MeshLib::Node(0.0, 0.0, 0.0);
+        nodes[1] = new MeshLib::Node(1.0, 0.0, 0.0);
+        nodes[2] = new MeshLib::Node(1.0, 1.0, 0.0);
+        nodes[3] = new MeshLib::Node(0.0, 1.0, 0.0);
+
+        nodes[4] = new MeshLib::Node(0.5, perturbation, 0.0);
```

Comment:
Yes, the analytical 3D volume with such perturbation is hard to obtain. I can try it for HEX20 first but it is not presented in this PR.